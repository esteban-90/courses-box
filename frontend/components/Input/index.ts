export * from './Input'
export * from './Feedback'
