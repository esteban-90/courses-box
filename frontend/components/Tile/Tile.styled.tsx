import styled from '@emotion/styled'
import { makeShadow } from '@/styles/helpers'

export const Wrapper = styled.section`
  padding: 1vmin 4vmin 4vmin;
  border-radius: 1rem;
  color: ${({ theme }) => theme.font.regular};
  background-color: ${({ theme }) => theme.background};

  box-shadow: ${({ theme }) => {
    const { shadow1, shadow2 } = theme.components
    const boxShadow = makeShadow(shadow1, shadow2)
    return boxShadow
  }};
`
