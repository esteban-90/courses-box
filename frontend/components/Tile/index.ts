export * from './Tile'
export * from './CenteredTile'
