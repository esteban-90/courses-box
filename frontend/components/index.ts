export * from './Button'
export * from './Checkbox'
export * from './Course'
export * from './Icon'
export * from './IconButton'
export * from './Input'
export * from './Layout'
export * from './Link'
export * from './Logo'
export * from './Switch'
export * from './Tile'
