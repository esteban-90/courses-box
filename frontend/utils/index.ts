import qs from 'qs'
import { apiUrl } from '@/config'
import type { CoursePayload, CoursesPayload, ErrorPayload, Course as CourseType } from '@/types'

const getData = async (param: string) => {
  const response = await fetch(`${apiUrl}/courses${param}`, { method: 'GET' })
  const result = await response.json()
  const { status } = response
  const error = status >= 400 ? (result as ErrorPayload) : undefined
  const success = status >= 200 && status < 400 ? (result as CoursesPayload | CoursePayload) : undefined

  return { data: success?.data, error: error?.error?.message ?? '' }
}

type Data<D> = Promise<{ data: D; error: string }>

export const getCourses = () => getData('?populate=*') as Data<CourseType[]>
export const getCourseById = (id: string) => getData(`/${id}?populate=*`) as Data<CourseType>

export const getCoursesByQuery = (q: string) => {
  const query = qs.stringify(
    {
      populate: '*',
      filters: {
        $or: [{ title: { $containsi: q } }, { subtitle: { $containsi: q } }, { description: { $containsi: q } }],
      },
    },
    { encodeValuesOnly: true }
  )

  return getData(`?${query}`) as Data<CourseType[]>
}
