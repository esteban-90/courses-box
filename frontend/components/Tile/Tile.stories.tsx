import { ComponentStoryObj, ComponentMeta } from '@storybook/react'
import { expect } from '@storybook/jest'
import { screen } from '@storybook/testing-library'
import { Tile } from '@/components/Tile'

export default {
  title: 'Content/Tile',
  component: Tile,
} as ComponentMeta<typeof Tile>

export const BasicTile: ComponentStoryObj<typeof Tile> = {
  async play() {
    await expect(screen.getByRole('heading')).toBeInTheDocument()
  },

  args: {
    heading: 'Lorem ipsum dolor sit amet.',
    children:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus ultricies scelerisque purus at egestas. Duis sagittis ullamcorper velit in vehicula. Aliquam odio sapien, maximus ac nisl id, sollicitudin pretium massa. Praesent eu vehicula diam. Fusce eu est urna. Phasellus iaculis, dui eu elementum imperdiet, sapien ante rutrum ligula, nec vulputate est lorem quis turpis. Pellentesque in velit in metus gravida interdum. Sed placerat orci at posuere pellentesque. Pellentesque at nulla quis enim malesuada efficitur. In hac habitasse platea dictumst. Nullam nisi justo, feugiat et dui nec, porta interdum elit. Morbi placerat nec magna nec aliquet. Proin facilisis nibh sit amet porttitor tempus. Duis et pellentesque magna. Integer sit amet magna pharetra, ultrices massa vitae, accumsan urna. Integer congue, nisl finibus euismod pulvinar, orci diam egestas diam, a hendrerit neque eros dignissim enim. Maecenas condimentum placerat sem sed iaculis. Sed et magna nulla. Praesent id purus in mauris rutrum consectetur. Sed accumsan mi vel enim ullamcorper gravida. Donec vel molestie nulla, in dignissim arcu. Phasellus nec libero vulputate orci cursus imperdiet. Nullam nec ex metus. Morbi maximus aliquam dui a mattis. Nulla faucibus, ipsum et lobortis iaculis, felis quam accumsan risus, a dignissim elit erat et nibh. Sed vestibulum, leo quis tempor congue, risus lorem tincidunt lacus, ut hendrerit purus sapien quis mauris. Integer molestie dui massa, at hendrerit magna convallis ornare. Suspendisse potenti. Cras pretium lectus quis mauris placerat molestie. Phasellus nec faucibus felis. Nullam et hendrerit quam. Nunc id ultricies ipsum. Proin ac placerat eros. Etiam metus nisl, pretium nec velit id, posuere feugiat turpis. Nulla facilisi. Pellentesque at posuere dolor.',
  },
}

export const BasicTileWithSmallAmountOfContent: ComponentStoryObj<typeof Tile> = {
  ...BasicTile,

  args: {
    heading: 'Lorem ipsum dolor sit amet',
    children: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
  },
}
