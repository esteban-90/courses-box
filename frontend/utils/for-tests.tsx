import { FC, ReactElement, ReactNode } from 'react'
import { Provider as StoreProvider } from 'react-redux'
import { configureStore } from '@reduxjs/toolkit'
import { render, RenderResult } from '@testing-library/react'
import { ThemeProvider } from '@emotion/react'
import { Layout } from '@/components'
import { rootReducer } from '@/store'
import { Themes } from '@/styles/themes'
import { StoreAndRenderOptions } from '@/types'

const customRender = (ui: ReactElement, { preloadedState, ...options }: StoreAndRenderOptions = {}): RenderResult => {
  const store = configureStore({ reducer: rootReducer, preloadedState })

  const Wrapper: FC<{ children: ReactNode }> = ({ children }): JSX.Element => {
    return (
      <StoreProvider store={store}>
        <ThemeProvider theme={Themes.light}>{children}</ThemeProvider>
      </StoreProvider>
    )
  }

  return render(ui, { wrapper: Wrapper, ...options })
}

const pageRender = (ui: ReactElement, { preloadedState, ...options }: StoreAndRenderOptions = {}): RenderResult => {
  const store = configureStore({ reducer: rootReducer, preloadedState })

  const Wrapper: FC<{ children: ReactNode }> = ({ children }): JSX.Element => {
    return (
      <StoreProvider store={store}>
        <Layout>{children}</Layout>
      </StoreProvider>
    )
  }

  return render(ui, { wrapper: Wrapper, ...options })
}

export * from '@testing-library/react'
export { customRender as render, pageRender }
