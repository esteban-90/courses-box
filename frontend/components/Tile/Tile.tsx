import type { FC, ReactNode } from 'react'
import * as Styled from '@styled/Tile'

export type TileProps = {
  /** Title */
  heading?: ReactNode
  /** Text content */
  children: ReactNode
}

export const Tile: FC<TileProps> = ({ heading, children, ...rest }): JSX.Element => {
  return (
    <Styled.Wrapper {...rest}>
      <h2>{heading}</h2>
      {children}
    </Styled.Wrapper>
  )
}
