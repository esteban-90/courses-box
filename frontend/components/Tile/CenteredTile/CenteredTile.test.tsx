import { CenteredTile } from '@/components/Tile'
import { render } from '@/tests/utils'

describe('Centered Tile test cases:', () => {
  it('should check render', () => {
    const centeredTileElement = (
      <CenteredTile heading='Lorem ipsum dolor sit amet'>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </CenteredTile>
    )

    const { asFragment } = render(centeredTileElement)
    expect(asFragment()).toMatchSnapshot()
  })
})
