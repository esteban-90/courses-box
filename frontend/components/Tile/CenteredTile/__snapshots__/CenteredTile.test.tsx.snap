// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Centered Tile test cases: should check render 1`] = `
<DocumentFragment>
  <div
    class="css-gbwc6v"
  >
    <section
      class="css-1c469nw"
    >
      <h2>
        Lorem ipsum dolor sit amet
      </h2>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </section>
  </div>
</DocumentFragment>
`;
